# Foundations of Data Analysis 1 - Week 1 - Cycling in Austin

<img src="images/week11.jpg">

In 2011, researchers at the Texas Transportation Institute and the Center for Transportation Research at UT Austin ran an advertising campaign aimed at recruiting Austin cyclists to join the South Congress Bike Mapping Project.  As members of the project, cyclists downloaded and used Cycle Tracks, a smartphone app developed by the San Francisco County Transportation Authority to track where people are riding their bikes based on their GPS points.  The goal was to gain new information about bike commuting patterns and this data set is based on the results of the study: 3600 trips tracked from 315 users over a 6 month period.   Data includes distances traveled, speed of travel, and reasons for travel among other variables.

In [1]:
BikeData <- read.csv("data/BikeData.csv")
head(BikeData)

user_id age gender student employed cyc_freq               distance time
1 1       28  M      1       1        Daily                  3.25     15  
2 2       35  M      0       1        Daily                  1.11      5  
3 3       28  M      0       1        Daily                  5.59     23  
4 4       44  F      0       1        Less than once a month 3.24     24  
5 5       42  M      0       1        Several times per week 7.81     26  
6 6       36  M      0       1        Several times per week 3.00     20  
  speed
1 13.00
2 13.32
3 14.58
4  8.10
5 18.02
6  9.00

In [2]:
str(BikeData)

'data.frame':	121 obs. of  9 variables:
 $ user_id : int  1 2 3 4 5 6 7 8 9 10 ...
 $ age     : int  28 35 28 44 42 36 45 54 39 44 ...
 $ gender  : Factor w/ 2 levels "F","M": 2 2 2 1 2 2 2 2 2 2 ...
 $ student : int  1 0 0 0 0 0 0 0 0 0 ...
 $ employed: int  1 1 1 1 1 1 1 1 1 0 ...
 $ cyc_freq: Factor w/ 4 levels "Daily","Less than once a month",..: 1 1 1 2 4 4 4 4 4 3 ...
 $ distance: num  3.25 1.11 5.59 3.24 7.81 ...
 $ time    : int  15 5 23 24 26 20 51 39 50 44 ...
 $ speed   : num  13 13.3 14.6 8.1 18 ...


What is the age of the 7th rider in the dataset?

In [3]:
BikeData$age[7]

[1] 45

How many of the first 10 riders in the dataset ride daily?

In [4]:
length(BikeData[BikeData$user_id<=10 & BikeData$cyc_freq=="Daily"])

[1] 3

What is the speed of the first female who cycles less than one time per month (in miles/hour)?

In [5]:
levels(BikeData$cyc_freq)

[1] "Daily"                   "Less than once a month" 
[3] "Several times per month" "Several times per week"

In [6]:
BikeData$speed[BikeData$cyc_freq=="Less than once a month" & BikeData$gender=='F']

[1]  8.1 14.8

What type of variable is student?

In [7]:
class(BikeData$student)

[1] "integer"

What type of variable is cyc_freq?

In [8]:
class(BikeData$cyc_freq)

[1] "factor"

What type of variable is distance?

In [9]:
class(BikeData$distance)

[1] "numeric"

How many students are in the dataset?

In [10]:
table(BikeData$student)


  0   1 
107  14 

Since we only want to work with the student data, let’s create a new data frame that only includes students.

In [11]:
student <-BikeData[BikeData$student==1,]
student

user_id age gender student employed cyc_freq               distance time
1     1     28  M      1       1        Daily                   3.25    15  
15   15     26  M      1       1        Daily                  10.94    34  
18   18     32  F      1       1        Daily                   9.34    40  
31   31     25  M      1       1        Daily                   1.25     6  
49   49     44  M      1       1        Several times per week  9.29    41  
58   58     26  F      1       1        Daily                   2.77    16  
59   59     44  M      1       1        Several times per week  4.84    22  
77   77     63  M      1       1        Several times per week  6.56    40  
86   86     22  M      1       0        Daily                   0.85     4  
88   88     25  M      1       1        Several times per week  6.07    29  
97   97     29  M      1       0        Several times per week  3.52    15  
103 103     25  F      1       0        Daily                  13.43    37  
111 111     30  M      1       0        Daily                   7.19    26  
115 115     33  F      1       1        Several times per week  8.31    42  
    speed
1   13.00
15  19.31
18  14.01
31  12.50
49  13.60
58  10.39
59  13.20
77   9.84
86  12.75
88  12.56
97  14.08
103 21.78
111 16.59
115 11.87

We want to know how often the students ride

In [12]:
table(student$cyc_freq)


                  Daily  Less than once a month Several times per month 
                      8                       0                       0 
 Several times per week 
                      6 

We also want to know how far the students travel on average. Let’s create a vector of just the distances

In [13]:
distance <-student$distance
distance

[1]  3.25 10.94  9.34  1.25  9.29  2.77  4.84  6.56  0.85  6.07  3.52 13.43
[13]  7.19  8.31

Now let’s find the average distance ridden by the students, using the mean function.

In [14]:
mean(distance)

[1] 6.257857

In this Pre Lab, we examined data on 14 student riders. Most of the student riders (a total of 8 out of 14) rode their bikes 
daily. On average, the students rode about 6.26 miles on each trip.

**How many of the cyclists rode daily, how many of these riders were male and how many female, and what what were their average ages?**

In [15]:
table(BikeData$cyc_freq)


                  Daily  Less than once a month Several times per month 
                     47                       2                      14 
 Several times per week 
                     58 

In [23]:
daily <- BikeData[BikeData$cyc_freq == "Daily",]
daily

user_id age gender student employed cyc_freq distance time speed
1     1     28  M      1       1        Daily     3.25    15   13.00
2     2     35  M      0       1        Daily     1.11     5   13.32
3     3     28  M      0       1        Daily     5.59    23   14.58
11   11     29  M      0       1        Daily     4.36    20   13.08
14   14     29  M      0       1        Daily     2.62    15   10.48
15   15     26  M      1       1        Daily    10.94    34   19.31
17   17     33  M      0       1        Daily    10.02    45   13.36
18   18     32  F      1       1        Daily     9.34    40   14.01
19   19     43  M      0       1        Daily     7.02    28   15.04
26   26     35  M      0       1        Daily     6.04    31   11.69
31   31     25  M      1       1        Daily     1.25     6   12.50
32   32     39  M      0       1        Daily     3.76    21   10.74
39   39     41  M      0       1        Daily     6.33    25   15.19
40   40     23  M      0       1        Daily     6.44    15   25.76
44   44     46  M      0       1        Daily     6.16    22   16.80
48   48     36  M      0       1        Daily     2.15    19    6.79
52   52     36  F      0       1        Daily     0.63     4    9.45
53   53     27  F      0       1        Daily     1.78     9   11.87
55   55     51  M      0       1        Daily     3.74    14   16.03
56   56     24  M      0       1        Daily     5.18    23   13.51
58   58     26  F      1       1        Daily     2.77    16   10.39
60   60     36  M      0       1        Daily     2.70    12   13.50
62   62     32  F      0       1        Daily    11.01    38   17.38
65   65     32  M      0       0        Daily     2.52     8   18.90
70   70     34  M      0       0        Daily    12.21    55   13.32
72   72     26  M      0       1        Daily    11.17    36   18.62
80   80     27  M      0       1        Daily     7.74    37   12.55
81   81     29  F      0       1        Daily     8.17    28   17.51
85   85     29  M      0       1        Daily     2.31    10   13.86
86   86     22  M      1       0        Daily     0.85     4   12.75
90   90     34  M      0       1        Daily     3.90    16   14.63
91   91     36  M      0       1        Daily     3.00    13   13.85
92   92     38  M      0       1        Daily     1.31     5   15.72
96   96     33  M      0       1        Daily     7.92    49    9.70
98   98     37  M      0       1        Daily     3.13    12   15.65
99   99     43  M      0       1        Daily     9.72    45   12.96
100 100     30  M      0       1        Daily     3.85    15   15.40
102 102     28  M      0       0        Daily     7.03    32   13.18
103 103     25  F      1       0        Daily    13.43    37   21.78
105 105     42  F      0       1        Daily     1.01     6   10.10
107 107     37  M      0       1        Daily    13.47    41   19.71
108 108     32  M      0       1        Daily     6.10    35   10.46
110 110     54  M      0       1        Daily     2.78    27    6.18
111 111     30  M      1       0        Daily     7.19    26   16.59
112 112     34  M      0       1        Daily     9.12    36   15.20
117 117     46  M      0       1        Daily     8.14    40   12.21
119 119     44  F      0       1        Daily     7.84    48    9.80

In [17]:
table(daily$gender)


 F  M 
 9 38 

In [18]:
mean(daily$age)

[1] 33.65957

In [19]:
daily_males <- daily[daily$gender == 'M',]
daily_females <- daily[daily$gender == 'F',]

In [20]:
mean(daily_males$age)
mean(daily_females$age)

[1] 33.92105

[1] 32.55556

**How many daily male riders are age 30 or older?**

In [21]:
thirty <- daily_males[daily_males$age >= 30,]
thirty

user_id age gender student employed cyc_freq distance time speed
2     2     35  M      0       1        Daily     1.11     5   13.32
17   17     33  M      0       1        Daily    10.02    45   13.36
19   19     43  M      0       1        Daily     7.02    28   15.04
26   26     35  M      0       1        Daily     6.04    31   11.69
32   32     39  M      0       1        Daily     3.76    21   10.74
39   39     41  M      0       1        Daily     6.33    25   15.19
44   44     46  M      0       1        Daily     6.16    22   16.80
48   48     36  M      0       1        Daily     2.15    19    6.79
55   55     51  M      0       1        Daily     3.74    14   16.03
60   60     36  M      0       1        Daily     2.70    12   13.50
65   65     32  M      0       0        Daily     2.52     8   18.90
70   70     34  M      0       0        Daily    12.21    55   13.32
90   90     34  M      0       1        Daily     3.90    16   14.63
91   91     36  M      0       1        Daily     3.00    13   13.85
92   92     38  M      0       1        Daily     1.31     5   15.72
96   96     33  M      0       1        Daily     7.92    49    9.70
98   98     37  M      0       1        Daily     3.13    12   15.65
99   99     43  M      0       1        Daily     9.72    45   12.96
100 100     30  M      0       1        Daily     3.85    15   15.40
107 107     37  M      0       1        Daily    13.47    41   19.71
108 108     32  M      0       1        Daily     6.10    35   10.46
110 110     54  M      0       1        Daily     2.78    27    6.18
111 111     30  M      1       0        Daily     7.19    26   16.59
112 112     34  M      0       1        Daily     9.12    36   15.20
117 117     46  M      0       1        Daily     8.14    40   12.21

**In this lab, we examined data on 47 daily riders. Most of the daily riders (a total of 38) were male. The average age of all daily riders was 33.7. The average age of female daily riders was 32.6, while the averge age of male daily riders was 
33.9.**